### PR TITLE
Remove propTypes definitions from Top Rated Products block

### LINKED ISSUE #9519: Convert `block.js` to `block.tsx` and replace `propTypes` with TypeScript definitions
## Problem

[`assets/js/blocks/product-top-rated/block.js`](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/assets/js/blocks/product-top-rated/block.js) is a JS file and uses `propTypes` definitions.

## Expected outcome

1. Convert the file from `block.js` to `block.tsx`.
2. Replace `propTypes` with TypeScript definitions.

### PR DESCRIPTION
**Pull request by hritikchaudhary**
Fixes #9519 

### Changes in the PR:
Removed propTypes definitions from product-search block and converted it to tsx.

#### Automated Tests
* [x] Changes in this PR are covered by Automated Tests.
  * [x] Unit tests
  * [x] E2E tests

#### User Facing Testing

* [x] Do not include in the Testing Notes 

### WooCommerce Visibility

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### CODE REVIEW

File: assets/js/blocks/product-top-rated/block.tsx [hunk 1]
```diff
@@ -6,18 +6,23 @@ import { Component } from '@wordpress/element';
...
+ * Internal dependencies
+ */
+import { Props } from './types';
+
 /**
  * Component to handle edit mode of "Top Rated Products".
  */
-class ProductTopRatedBlock extends Component {
+
+class ProductTopRatedBlock extends Component< Props > {
```
Comment (nielslange) [inline assets/js/blocks/product-top-rated/block.tsx]: Initially, I didn't notice that this file was still a class component. When working on files that are class components, we usually convert them to functional components. Do you think you could that in this case? If so, we should also adjust `getInspectorControls()`, in the line below, accordingly.

**Review by nielslange — CHANGES_REQUESTED** [review]
Thank you so much for working on this, @hritikchaudhary. I just added a comment regarding converting the file from a class component into a functional component.

I also noticed that this change introduces two TS errors in the following files:
- `assets/js/blocks/product-top-rated/block.tsx`
- `assets/js/blocks/product-top-rated/index.js`

The TS error in `assets/js/blocks/product-top-rated/block.tsx` is related to `import ServerSideRender from '@wordpress/server-side-render';` and can be ignored.

The TS error in `assets/js/blocks/product-top-rated/index.js` is related to overloading the `registerBlockType()` call. Could you look into this and also convert that file to TS?

**Review by nielslange — CHANGES_REQUESTED** [review]
Thank you so much for working on this, @hritikchaudhary. I just added a comment regarding converting the file from a class component into a functional component.

I also noticed that this change introduces two TS errors in the following files:
- `assets/js/blocks/product-top-rated/block.tsx`
- `assets/js/blocks/product-top-rated/index.js`

The TS error in `assets/js/blocks/product-top-rated/block.tsx` is related to `import ServerSideRender from '@wordpress/server-side-render';` and can be ignored.

The TS error in `assets/js/blocks/product-top-rated/index.js` is related to overloading the `registerBlockType()` call. Could you look into this and also convert that file to TS?

File: assets/js/blocks/product-top-rated/block.tsx (hunk 1 above)
Comment (hritikchaudhary) [reply assets/js/blocks/product-top-rated/block.tsx]: Hi sure I'll work on it... I was not aware about that.
Not sure how `assets/js/blocks/product-top-rated/index.js` got flagged though, I didn't make any change in it, anyway will look into it.
Thanks for the feedback and direction!

**Comment by hritikchaudhary** [comment]
Hi @nielslange 

1. I have converted class components to functional components
2. To resolve issues and convert index.js file, I had to take reference from product-best-sellers block as there were multiple errors while resolving overloading the registerBlockType() call isssue and overloading issue is still there but it is no longer flagging the entire file and I checked other components as well which are using metadata approach.. Let me know if there is a different approach I can follow.
4. Sorry it took some time to push the changes as I primarily use angular for frontend at my current job, so I'm learning at the same time 😀

**Comment by nielslange** [comment]
> 1. I have converted class components to functional components

Fantastic!

> 2. To resolve issues and convert index.js file, I had to take reference from product-best-sellers block as there were multiple errors while resolving overloading the registerBlockType() call isssue and overloading issue is still there but it is no longer flagging the entire file and I checked other components as well which are using metadata approach.. Let me know if there is a different approach I can follow.

Your approach is absolutely fine. 

> 3. Sorry it took some time to push the changes as I primarily use angular for frontend at my current job, so I'm learning at the same time 😀

Don't worry, @hritikchaudhary. Please take all the time that you need. After all, you are contributing in your spare time, which we appreciate very much. It's also great to hear that use this opportunity to learn React. That's fantastic. I'm sure you can find a lot of interesting code patterns within WooCommerce Blocks.

**Review by nielslange — APPROVED** [review]
Thanks for applying the suggestions, @hritikchaudhary. The changes look great, and I'll merge this PR now. Thanks again for working on this. 🙏

**Pull request merged**